cources/src/app/update-item/update-item.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import {CourcesService} from '../cources.service';
+import { ICourceItem } from '../icourceitem';
+import { TouchSequence } from 'selenium-webdriver';
+import {Router} from "@angular/router"
+import { CourceItem } from '../cource-item';
+
+@Component({
+  selector: 'app-update-item',
+  templateUrl: './update-item.component.html',
+  styleUrls: ['./update-item.component.css'],
+  providers: [CourcesService]
+})
+export class UpdateItemComponent implements OnInit {
+
+  private id: number;
+  private sub: any;
+  public isNew: boolean
+  public courceItem : ICourceItem;
+  public pageTitle : string;
+  constructor(private route: ActivatedRoute, private courcesService: CourcesService, private router: Router) { 
+  }
+
+  ngOnInit() {
+    this.sub = this.route.params.subscribe(params => {
+      this.id = +params['id']; 
+      this.isNew = false;
+      if(this.id > 0)
+      {
+       this.courceItem = this.courcesService.getCourceById(this.id);
+       this.pageTitle = "Update cource item";
+      }
+      else
+      {
+       this.courceItem = new CourceItem();
+       this.isNew = true;
+       this.pageTitle = "Add cource item";
+      }
+   });
+  }
+
+  update(isNew:boolean)
```

Comment:
ogólnie sprawdzenie czy to jest nowy item czy update może być wykonane w sevice. wystarczy sprawdzić czy ma ID. Wtedy servisw podejmie decyzję co zrobić. Zapisać nowy czy udate.

Reply:
czyli nie potrzebujesz tego `isNew`